homework/Vasili_Zasinets/Shape/src/main/java/com/tutrit/java/quickstart/shape/AbstractShape.java

```diff
@@ -0,0 +1,18 @@
+package com.tutrit.java.quickstart.shape;
+
+public abstract class AbstractShape implements Shape {
+    protected String name;
+
+    protected AbstractShape(String name) {
+        this.name = name;
+    }
+
+    public abstract double circumference() throws ShapeLengthException;
+
+    public abstract double perimeter() throws  ShapeLengthException;
```

Comment:
circumference and perimetr is the synonyms.. in a varying degree ...and Why are they here?
In description task will write create method getArea... Where is it?

Reply:
Fixed with area calculation. And the perimeter and long circle question is more in the field of mathematics than programming.